apisix/plugins/server-info.lua

```diff
@@ -158,25 +159,65 @@ local function report(premature, report_ttl)
         end
     end
 
-    server_info.last_report_time = ngx_time()
-
+    -- get inside etcd data, if not exist, create it
     local key = "/data_plane/server_info/" .. server_info.id
-    local ok, err = core.etcd.set(key, server_info, report_ttl)
+    local res, err = core.etcd.get(key)
+    if err ~= nil then
+        core.log.error("failed to get server_info from etcd: ", err)
```

Comment:
if `err` is not `nil`, why can we go ahead?

Reply:
When a apisix node be  initialed, the ETCD don't have server-info data, I think the `get` func will report error, so I continue the code and set the info to ETCD. Maybe use `if not res` be better.

Reply:
No, if `get` method reports error when no data, you should check the error type and decide whether returning or going ahead.